eng/scripts/Language-Settings.ps1

```diff
@@ -5,31 +5,117 @@ $packagePattern = "*.zip"
 $MetadataUri = "https://raw.githubusercontent.com/Azure/azure-sdk/master/_data/releases/latest/python-packages.csv"
 $BlobStorageUrl = "https://azuresdkdocs.blob.core.windows.net/%24web?restype=container&comp=list&prefix=python%2F&delimiter=%2F"
 
-function Get-python-PackageInfoFromRepo  ($pkgPath, $serviceDirectory, $pkgName)
-{
-  pip install packaging==20.4 -q -I
-  $pkgName = $pkgName.Replace('_', '-')
-  if (Test-Path (Join-Path $pkgPath "setup.py"))
+function Get-python-PackageInfoFromRepo  ($pkgPath, $serviceDirectoryName, $artifactName)
+{  
+  $packageName = $artifactName.Replace('_', '-')
+  $pkgDirName = Split-Path $pkgPath -Leaf
+  if ($pkgDirName -ne $packageName)
+  {
+    # Common code triggers this function against each directory but we can skip if it doesn't match package name
+    return $null
+  }
+
+  $setupPyPath = Join-Path $pkgPath "setup.py"
```

Comment:
Is this PR just an experiment to compare performance? Or is this something you discussed with @scbedd and @praveenkuttappan as a better option to make this all work.

Reply:
I started out thinking it will be just a few checks on a few files. Then it turns out there were many inconsistencies with version and package Name is specified. I will go ahead and schedule meeting with Praveen to investigate a better way to do this. The performance for Get-AllPackages was improved from several minutes to 15 secs